src/components/Nav.js

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import MenuIcon from "../assets/menu.png";
+import nextId from "react-id-generator";
+
+class Nav extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isShowingMobile: false,
+    };
+
+    this.showLinks = this.showLinks.bind(this);
+    this.displayDesktopProtectedRoutesIfAuthenticated = this.displayDesktopProtectedRoutesIfAuthenticated.bind(
+      this
+    );
+    this.displayMobileProtectedRoutesIfAuthenticated = this.displayMobileProtectedRoutesIfAuthenticated.bind(
+      this
+    );
+  }
+
+  componentDidUpdate() {
+    //Show menu in mobile
+    const mobileNav = document.querySelector(".mobile-nav-right-container");
+
+    if (!mobileNav.classList.contains("hide")) {
+      mobileNav.classList.add("hide");
+    } else {
+      mobileNav.classList.remove("hide");
+    }
+  }
+
+  showLinks() {
+    this.setState((state, props) => {
+      return {
+        isShowingMobile: !state.isShowingMobile,
+      };
+    });
+  }
+
+  displayMobileProtectedRoutesIfAuthenticated() {
+    const { isAuthenticated } = this.props;
+    if (isAuthenticated) {
+      return [
+        <li>
+          <Link to="/journals">Journals</Link>
+        </li>,
+        <li>
+          <Link to="/strategies">Strategies</Link>
+        </li>,
+        <li key={nextId()}>
+          <Link onClick={() => this.props.logout()} to="/">
+            Log out
+          </Link>
+        </li>,
+      ];
+    }
+
+    return [
+      <li key={nextId()}>
+        <Link to="/login">Log in</Link>
+      </li>,
+      <li key={nextId()}>
+        <Link to="/register">Register</Link>
+      </li>,
+    ];
+  }
+
+  displayDesktopProtectedRoutesIfAuthenticated() {
```

Comment:
I think the desktop and mobile routes should be the same. If that's the case, we should only have one function to display the appropriate routes `displayRoutes`.